modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/slot/Slot.java

```diff
@@ -52,6 +61,16 @@
     private boolean isSlotActive;
 
     /**
+     * Indicates whether the slot is a fresh one or an overlapped one
+     */
+    private boolean isAnOverlappingSlot;
```

Comment:
If the order of states saved (ArrayList) doesn't matter we can use a HashSet for States and use HashSet#contains to have the same behaviour. Just an Idea.